zeebe/broker-client/src/main/java/io/camunda/zeebe/broker/client/impl/BrokerRequestManager.java

```diff
@@ -230,6 +240,13 @@ private BrokerAddressProvider determineBrokerNodeIdProvider(final BrokerRequest<
     }
   }
 
+  private void throwIfPartitionInactive(final int partitionId) {
+    final BrokerClusterState topology = topologyManager.getTopology();
+    if (topology != null && topology.getInactiveNodesForPartition(partitionId) != null) {
+      throw new PartitionInactiveException(partitionId);
+    }
+  }
```

Comment:
:x: This is will throw even if only one replica is Inactive, but the other two replicas are healthy and there is a leader. To be more precise we can do something like : "throw if there is atleast one inactive replica AND there is no healthy leader". 

The chances of only one replica becoming inactive is very less, but we shouldn't unnecessarily reject a request if there is a healthy leader.

Reply:
Also `topology.getInactiveNodesForPartition(partitionId) != null` this will be always true, because the returned set is never null. You should check if the set is empty.

Reply:
I'll also write a test for this case.

Reply:
> Also `topology.getInactiveNodesForPartition(partitionId) != null` this will be always true, because the returned set is never null. You should check if the set is empty.

I thought it is possible to be null because `partitionInactiveNodes` is a map and the `.get` call can return null, also for some `BrokerClientTest` the return value  from `.getInactiveNodesForPartition(id)` will be null 
<img width="722" alt="Screenshot 2024-07-01 at 09 52 59" src="https://github.com/camunda/camunda/assets/168723167/feb04a48-ba3f-48ec-8968-c5456b5c7994">

<img width="2050" alt="image" src="https://github.com/camunda/camunda/assets/168723167/492dbd81-5ef3-49d5-8f14-131b4bd7e8e8">